include/deal.II/grid/grid_tools.h

```diff
@@ -2651,6 +2654,35 @@ namespace GridTools
     const std::vector<BoundingBox<spacedim>> &local_bboxes,
     MPI_Comm                                  mpi_communicator);
 
+  template <int spacedim>
+  using Rtree = boost::geometry::index::rtree<
+    std::tuple<BoundingBox<spacedim>, unsigned int>,
+    boost::geometry::index::linear<16>>;
```

Comment:
I've looked at this patch again, and I'm not sure I'm in favour of adding this global `Rtree` declaration. `Rtree` is a general name, but this is a very specific type, only used in one function, and most likely not useful for other purposes where and `Rtree` would also be useful. 

Maybe it would make sense to expose a more general `Rtree` alias (and *document* it.... there is no documentation whatsoever of what `Rtree` is here, what algorithm uses, etc.). And with the correct `CamelCase` name: `RTree`, i.e., 
~~~
template <typename LeafType>
  using RTree = boost::geometry::index::rtree<
    LeafType,
    boost::geometry::index::linear<16>>;
~~~
and provide a function like
~~~
template<typename LeafTypeIterator> 
Rtree<typename LeafTypeIterator::value_type>
pack_rtree(LeafTypeIterator begin, LeafTypeIterator end);
~~~
to automatically generate the right `RTree`.